apps/src/blockly/addons/cdoSerializationHelpers.js

```diff
@@ -181,3 +185,19 @@ function reorderBlocks(blocks, blockOrderMap) {
 
   return orderedBlocks;
 }
+
+function sortProceduresAfterSetup(blockA, blockB) {
+  const blockAType = blockA.type;
+  const blockBType = blockB.type;
+
+  const blockAIsSetup = SETUP_TYPES.includes(blockAType);
+  const blockBIsProcedure = PROCEDURE_DEFINITION_TYPES.includes(blockBType);
+
+  if (blockAIsSetup && blockBIsProcedure) {
```

Comment:
It would be worth adding a comment to this method explaining it a bit more. If I understand correctly, it puts all setup blocks before all procedure blocks, but then other blocks can just be anywhere?

Reply:
Yeah, and I don't think procedure blocks will always come before other block types, i.e., conditions #2 and #3 (pasted below for clarity) don't necessarily need to be handled separately.  For instance, if blockA is neither setup nor procedure and blockB is a procedure block, the function will return 1, indicating that blockB (procedure) should come before blockA (other).
```
else if (!blockAIsSetup && !blockBIsProcedure) {
  return 0; // no changes in order
} else {
  return 1; // blockB comes before blockA or no changes in order
}
```
Can you confirm that's your understanding? I'm thinking the clearest way to rewrite this would be something that just clearly moves setup blocks to the front.

Reply:
Resolved offline.